vgen/src/main/java/io/gapi/vgen/InterfaceConfig.java

```diff
@@ -40,7 +40,7 @@
  * configuration for methods and resource names.
  */
 public class InterfaceConfig {
-  private final ImmutableList<CollectionConfig> collectionConfigs;
+  private final Map<String, CollectionConfig> collectionConfigs;
```

Comment:
I see that we're using ImmutableMap for some maps here, but Map for others. Maybe we should use ImmutableMap for them all?